Assets/Shopify/UIToolkit/Components/Dropdown/OptionDropdown.cs

```diff
@@ -0,0 +1,111 @@
+namespace Shopify.UIToolkit {
+    using UnityEngine;
+    using UnityEngine.UI;
+    using UnityEngine.EventSystems;
+    using UnityEngine.Events;
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+
+    public class OptionDropdown : MonoBehaviour {
+        public string[] Options;
+        public string Title;
+
+        public GameObject TitleLabel;
+        public GameObject ScrollViewContent;
+        public GameObject SelectedOptionPrefab;
+        public GameObject UnselectedOptionPrefab;
+        public UnityEvent onSelectOption = new UnityEvent();
+        public DropdownOptionItem SelectedItem {
+            get {
+                var child = ScrollViewContent.transform.GetChild(SelectedIndex);
+                var optionObj = child.transform.GetChild(0);
+                return optionObj.GetComponent<DropdownOptionItem>();
+            }
+        }
+
+        private int SelectedIndex;
+
+        public void Build() {
+            TitleLabel.GetComponent<Text>().text = Title;
+
+            // Configure the dropdown to contain options.
+            for (var i = 0; i < Options.Length; i++) {
+                var option = Options[i];
+                AddOptionNamed(option, i == SelectedIndex);
+            } 
+
+            // Resize the scroll view content view to fit the options snug.
+            var contentRectTransform = ScrollViewContent.GetComponent<RectTransform>();
+            contentRectTransform.sizeDelta 
+                = new Vector2(contentRectTransform.rect.width, CalculateHeightOfContentChildren());
+        }
+
+        public void Clear() {
+            foreach (Transform child in ScrollViewContent.transform) {
+                GameObject.Destroy(child.gameObject);
+            }
+        }
+
+        public void SelectOption() {
+            var newSelection = EventSystem.current.currentSelectedGameObject;
+            var previousSelection = ScrollViewContent.transform.GetChild(SelectedIndex);
+
+            // Don't bother reselecting a the previous selection!
+            if (newSelection == previousSelection.gameObject) {
+                return;
+            }
+
+            // Deselect previous selection and new selection containers.
+            foreach (Transform child in previousSelection.transform) {
+                GameObject.Destroy(child.gameObject);
```

Comment:
you sure you want to destroy here? The GC will have to cleanup... we can also recycle these views but if you opt for that i think doing it in another pr would be best

Reply:
Ideally we'd do recycling but for now I figured we'd just throw it away. I don't think we'll have too many dropdown items anyways but I can profile it just in case.

A good follow up PR would be to add in list recycling that we can also use for product listings.

@richardmonette Do we already have code that does this as part of the example app?

Reply:
In the example app right now we combine all the options (there is just one selector with all the combinations). It could be upgraded to this though.